account_invoice_tier_validation/models/account_invoice.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2020 ForgeFlow S.L.
+# Copyright 2021 Onestein (<https://www.onestein.eu>)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl).
+
+import threading
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _name = "account.invoice"
+    _inherit = ["account.invoice", "tier.validation"]
+    _state_from = ["draft"]
+    _state_to = ["open", "paid"]
+
+    @api.model
+    def _get_under_validation_exceptions(self):
+        res = super()._get_under_validation_exceptions()
+        if self.env.context.get("invoice_tier_validation_allow_write"):
+            res += ["move_id", "date", "move_name", "date_due", "date_invoice"]
```

Comment:
```suggestion
            res += ["move_id", "date", "move_name", "date_due", "date_invoice", "reference"]
```

I suppose this should fix this issue https://github.com/OCA/account-invoicing/pull/871#issuecomment-791408920, although I haven't checked the record in runbot.  I got the same error in my local environment and the reference field was the cause.